### PR TITLE
Fix issues after running static analysis: part 5

### PR DESCRIPTION
**Pull request by amiclaus**
Fifth set of fixed issues after running cppcheck tool.

Signed-off-by: Antoniu Miclaus antoniu.miclaus@analog.com

### CODE REVIEW

**Comment by amiclaus** [comment]
Changelog: v1->v2:
* added "Signed-off by:" to each commit.
* added description to each commit.

File: fmcadc5/fmcadc5.c [hunk 1]
```diff
@@ -198,6 +198,7 @@ int main(void)
 	i5g_init_param.ad9625_cs_0 = 1;
 	i5g_init_param.ad9625_cs_1 = 2;
 	i5g_init_param.regs = XPAR_AXI_FMCADC5_SYNC_BASEADDR;
+	i5g_init_param.sysref_delay = 0;
```
Comment (commodo) [inline fmcadc5/fmcadc5.c]: this is good;
another way to do it, is to do a `memset(&i5g_init_param, 0, sizeof(i5g_init_param))` right before these assignments ; 
this typically makes sure that all memory is 0, except for the parameters which are specifically initialized;

File: common_drivers/fru_tools/fru_tools.c [hunk 2]
```diff
@@ -610,6 +610,7 @@ static unsigned int insert_str(unsigned char *buf, const unsigned char * str, bo
 			}
 			buf[0] = tmp | (FRU_STRING_ASCII << 6);
 			memcpy(&buf[1], &str[1], tmp);
+			free(six);
 			return tmp + 1;
 		}
```
Comment (commodo) [inline common_drivers/fru_tools/fru_tools.c]: i think we may also need to free this after line 615, to make sure we don't leak in case `tmp >= 0`

Comment (commodo) [reply common_drivers/fru_tools/fru_tools.c]: my bad;
i didn't see it at line 623

File: fmcadc5/fmcadc5.c (hunk 1 above)
Comment (commodo) [reply fmcadc5/fmcadc5.c]: ack: disregard my comment :)

**Comment by commodo** [comment]
LGTM

**Comment by commodo** [comment]
Will merge later today if no objections

**Pull request merged**